### DISCUSSION TITLE
Video "PyTorch for Deep Learning & Machine Learning – Full Course"- Coding error in tensor-like

### DISCUSSION THREAD

**Discussion by Rohi2610** [mrdbourke/pytorch-deep-learning#914] [Q&A] [answered]
Hello Sir, I'm a student trying to learn PyTorch from your amazing YouTube video. As you suggested, I've been experimenting around with almost every topic that's being discussed. So after learning Tensor-like, I tried creating a Torch.rand_like. What happened here is that if I use torch.range() to create a tensor and then create a tensor-like, the code works smoothly.

Code:
```python
one_ten= torch.range(1, 10)
ten_rand= torch.rand_like(input=one_ten) 
ten_rand
```

The error happens when I use torch.arange() or any tensor created using arange. I have attached the error as a picture. It basically says  "check_uniform_bounds" not implemented for 'Long'. I tried changing the datatype of the tensor to float as some articles in Google and the collab AI suggested but it was no good. Below is the code with torch.arange(). 

```python
one_ten= torch.arange(1, 10)
one_ten.float
ten_rand= torch.rand_like(input=one_ten) 
ten_rand
```
<img width="581" alt="Screenshot 2024-05-01 131649" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/144462900/9997c708-60f9-45de-8186-9cba7befb6c7">

**Comment by LuluW8071** [comment, marked as answer]
You need to change `torch.dtype()` when using `torch.arange()` since in the deprecated (old version) of Torch, when using `torch.range()`, the values were __float32__ by default. Now, for `torch.arange()`, its dtype is __Long__ by default.

So, looking at your `RuntimeError: "check_uniform_bounds" not implemented for 'Long'` error, `one_ten` generated by `torch.arange()` yields **Long** values, but `torch.rand_like` only accepts **float** values. To debug this, simply change the dtype.

```python
one_ten= torch.arange(1, 10).to(torch.float32)
ten_rand= torch.rand_like(input=one_ten) 
ten_rand
```

You can also do `print(one_ten)` on both `range` and `arange` to know the difference and check their dtypes.

**Reply by Rohi2610** [reply]
Thank you so much. It worked and I understood the problem.
